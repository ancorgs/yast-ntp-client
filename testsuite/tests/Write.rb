# encoding: utf-8

# File:
#  Read.ycp
#
# Module:
#  NTP client configurator
#
# Summary:
#  Reading configuration testsuite
#
# Authors:
#  Jiri Srain <jsrain@suse.cz>
#
# $Id$
#
module Yast
  class WriteClient < Client
    def main
      # testedfiles: NtpClient.ycp

      Yast.include self, "testsuite.rb"

      @READ = {
        "init"      => { "scripts" => { "exists" => true } },
        "etc"       => {
          "ntp_conf" => {
            "all" => {
              "comment" => "",
              "file"    => -1,
              "kind"    => "section",
              "name"    => "",
              "type"    => -1,
              "value"   => []
            }
          }
        },
        "sysconfig" => {
          "ntp"               => { "NTPD_RUN_CHROOTED" => "yes" },
          "personal-firewall" => { "REJECT_ALL_INCOMING_CONNECTIONS" => "no" }
        },
        "target"    => { "string" => "", "tmpdir" => "/tmp", "size" => 0 }
      }
      @WRITE = {}
      @EXEC = {
        "target" => {
          "bash_output" => {
            "exit" => 0,
            "stdout" => "",
            "stderr" => ""
          }
        }
      }


      TESTSUITE_INIT([@READ, @WRITE, @EXEC], nil)

      Yast.import "Progress"
      Yast.import "NtpClient"
      Yast.import "Mode"

      Mode.SetTest("testsuite")

      @progress_orig = Progress.set(false)

      TEST(lambda do
        NtpClient.Import(
          {
            "peers"           => [
              {
                "address"      => "ntp1",
                "initial_sync" => true,
                "options"      => " options1",
                "type"         => "server"
              },
              {
                "address"      => "127.127.1.2",
                "initial_sync" => false,
                "options"      => " options_clock",
                "type"         => "__clock"
              },
              {
                "address"      => "ntp2",
                "initial_sync" => true,
                "options"      => " options2",
                "type"         => "server"
              },
              {
                "address"      => "peer1",
                "initial_sync" => false,
                "options"      => " options_peer",
                "type"         => "peer"
              }
            ],
            "start_at_boot"   => false,
            "start_in_chroot" => true,
            "configure_dhcp"  => false
          }
        )
      end, [
        @READ,
        @WRITE,
        @EXEC
      ], nil)


      TEST(lambda { NtpClient.Write }, [@READ, @WRITE, @EXEC], nil)

      nil
    end
  end
end

Yast::WriteClient.new.main
