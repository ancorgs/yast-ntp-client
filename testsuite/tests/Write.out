Return	true
Read	.etc.ntp_conf.all $["comment":"", "file":-1, "kind":"section", "name":"", "type":-1, "value":[]]
Write	.etc.ntp_conf.all $["comment":"", "file":-1, "kind":"section", "name":"", "type":-1, "value":[$["comment":"", "kind":"value", "name":"server", "type":0, "value":"ntp1  options1"], $["comment":"", "kind":"value", "name":"server", "type":0, "value":"127.127.1.2  options_clock"], $["comment":"", "kind":"value", "name":"fudge", "type":0, "value":"127.127.1.2 "], $["comment":"", "kind":"value", "name":"server", "type":0, "value":"ntp2  options2"], $["comment":"", "kind":"value", "name":"peer", "type":0, "value":"peer1  options_peer"], $["comment":"", "kind":"value", "name":"restrict", "type":0, "value":"-6  default kod notrap nomodify nopeer noquery"], $["comment":"\n# Local users may interrogate the ntp server more closely.\n", "kind":"value", "name":"restrict", "type":0, "value":"127.0.0.1  "], $["comment":"", "kind":"value", "name":"restrict", "type":0, "value":"::1  "]]] true
Write	.etc.ntp_conf nil true
Write	.sysconfig.network.config.NETCONFIG_NTP_POLICY "" true
Write	.sysconfig.network.config nil true
Execute	.target.bash "/sbin/netconfig update -m ntp" 0
Write	.sysconfig.ntp.NTPD_RUN_CHROOTED "yes" true
Write	.sysconfig.ntp nil true
Execute	.target.bash "test -e /etc/cron.d/novell.ntp-synchronize  && rm /etc/cron.d/novell.ntp-synchronize;" 0
Return	true
